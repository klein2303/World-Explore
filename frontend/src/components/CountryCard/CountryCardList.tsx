import styles from "./CountryCardList.module.css";
import CountryCard from "./CountryCard";
import {CountryType} from "../../types/CountryTypes";

interface CountryCardListProps {
    countries: CountryType[];
}

const CountryCardList = ({countries}: CountryCardListProps) => {
    return (
        <main className={styles.cardlist} role="list">
            {countries.map((country: CountryType) => (
                <div role="listitem" key={country.name} className={styles.listitem}>
                    <CountryCard name={country.name} image={country.image} />
                </div>
            ))}
        </main>
    );
};

export default CountryCardList;
