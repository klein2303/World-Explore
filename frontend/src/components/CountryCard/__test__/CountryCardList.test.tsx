import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import { describe, it, expect } from "vitest";
import CountryCardList from "../CountryCardList";
import { MemoryRouter } from "react-router-dom";

describe("CountryCardList Tests", () => {
    it("renders the country card list correctly", () => {
        // Mock data
        const countries = [
            {
                name: "Australia",
                continent: "Australia",
                capital: "Cair Paravel",
                largest_city: "Atlantis Metro",
                currency: "Elbo",
                language: "Atlantean",
                population: 21299617,
                land_area: 599550.56,
                agriculture_area: "40%",
                forest_area: "25%",
                co2_emissions: 12.15,
                image: "https://example.com/australia.jpg",
            },
            {
                name: "Brazil",
                continent: "Africa",
                capital: "Cair Paravel",
                largest_city: "Atlantis Metro",
                currency: "Elbo",
                language: "Atlantean",
                population: 21299617,
                land_area: 599550.56,
                agriculture_area: "40%",
                forest_area: "25%",
                co2_emissions: 12.15,
                image: "https://example.com/brazil.jpg",
            },
        ];

        // Render the component
        const { asFragment } = render(
            <MemoryRouter>
                <CountryCardList />
            </MemoryRouter>,
        );

        // Take a snapshot of the initial render
        expect(asFragment()).toMatchSnapshot();

        // Check if the country cards are in the document
        expect(screen.getByText("Australia")).toBeInTheDocument();
        expect(screen.getByText("Brazil")).toBeInTheDocument();

        // Check if the images are in the document
        const australiaImage = screen.getByAltText("image of the country Australia");
        expect(australiaImage).toBeInTheDocument();
        expect(australiaImage).toHaveAttribute("src", "https://example.com/australia.jpg");

        const brazilImage = screen.getByAltText("image of the country Brazil");
        expect(brazilImage).toBeInTheDocument();
        expect(brazilImage).toHaveAttribute("src", "https://example.com/brazil.jpg");
    });
});
