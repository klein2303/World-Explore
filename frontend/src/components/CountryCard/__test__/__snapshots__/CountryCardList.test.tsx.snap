// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CountryCardList Tests > renders the country card list correctly 1`] = `
<DocumentFragment>
  <div
    class="_cardlist_beaa0e"
    role="list"
  >
    <div
      role="listitem"
    >
      <div
        aria-labelledby="country-name-Australia"
        class="_cardcontainer_043c24"
        role="region"
      >
        <img
          alt="image of the country Australia"
          aria-hidden="true"
          class="_cardimage_043c24"
          src="https://example.com/australia.jpg"
        />
        <div
          class="_information_043c24"
        >
          <p
            class="_name_043c24"
            id="country-name-Australia"
          >
            Australia
          </p>
          <svg
            aria-label="Mark Australia as visited"
            aria-pressed="false"
            class="_icon_043c24"
            fill="none"
            height="1em"
            role="button"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M20 10c0 6-8 12-8 12s-8-6-8-12a8 8 0 0 1 16 0Z"
            />
            <circle
              cx="12"
              cy="10"
              r="3"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      role="listitem"
    >
      <div
        aria-labelledby="country-name-Brazil"
        class="_cardcontainer_043c24"
        role="region"
      >
        <img
          alt="image of the country Brazil"
          aria-hidden="true"
          class="_cardimage_043c24"
          src="https://example.com/brazil.jpg"
        />
        <div
          class="_information_043c24"
        >
          <p
            class="_name_043c24"
            id="country-name-Brazil"
          >
            Brazil
          </p>
          <svg
            aria-label="Mark Brazil as visited"
            aria-pressed="false"
            class="_icon_043c24"
            fill="none"
            height="1em"
            role="button"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M20 10c0 6-8 12-8 12s-8-6-8-12a8 8 0 0 1 16 0Z"
            />
            <circle
              cx="12"
              cy="10"
              r="3"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
